### PR TITLE
wifi: monitor: Wireshark decode support

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

**Review by VivekUppunda — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

File: samples/wifi/monitor/README.rst [hunk 1]
```diff
@@ -117,6 +117,68 @@ Testing
...
+The packets from the device are sent to the host over the net capture tunnel using IPIP tunneling.
+The nRF70 Series device sends the IEEE 802.11 packets prepended with a custom header (for meta-data) over the net capture tunnel to the host.
+To analyze the packets in Wireshark, the payload of the UDP packets must be dissected as IEEE 802.11 packets.
+
+This support is only available in Wireshark 4.3 (under development).
+A custom build of Wireshark from latest sources is required.
+Once the custom build is installed, the following steps are required to dissect the payload of the UDP packets as IEEE 802.11 packets:
+
+#. Ensure Wireshark is compiled with Lua support.
+#. Open Wireshark and go to `Analyze` -> `Decode As...` -> `+` -> `UDP`.
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:141]: Assuming they are UI elements.
```suggestion
#. Open Wireshark and go to **Analyze** > **Decode As...** > **+**, then select **UDP**.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: samples/wifi/monitor/README.rst [hunk 2]
```diff
@@ -126,6 +126,61 @@ To enable this feature, use the configuration overlay files :file:`overlay-net-c
 When the offline net capture feature is enabled, incoming IEEE 802.11 packets are routed to the offline storage over the net capture tunnel.
 These packets can then be analyzed using Wireshark.
 
+Wireshark decode as IEEE 802.11
+===============================
+
+The packets from the device are sent to the host over the net capture tunnel using IP in IP tunneling.
+The nRF70 Series device sends the IEEE 802.11 packets prepended with a custom heametadata) over the net capture tunnel to the host.
```
Comment (kittydepa) [inline samples/wifi/monitor/README.rst:133]:
```suggestion
The nRF70 Series device sends the IEEE 802.11 packets prepended with a custom metadata over the net capture tunnel to the host.
```

File: samples/wifi/monitor/README.rst [hunk 3]
```diff
@@ -126,6 +126,61 @@ To enable this feature, use the configuration overlay files :file:`overlay-net-c
...
+
+         -- Register the dissector
+         local netcapture_udp_port = 4242
+         local udp_port = DissectorTable.get("udp.port")
+         udp_port:add(netcapture_udp_port, nordic_raw_80211)
+
+#. Copy the Lua script to the Wireshark plugin directory.
+     The plugin directory can be found in the Wireshark preferences.
+#. Open Wireshark and either start capturing packets or open a capture file.
+#. The UDP payload for port ``4242`` is now dissected as :
```
Comment (kittydepa) [inline samples/wifi/monitor/README.rst:179]:
```suggestion
#. The UDP payload for port ``4242`` is now dissected as the following:
```

File: samples/wifi/monitor/README.rst [hunk 4]
```diff
@@ -126,6 +126,61 @@ To enable this feature, use the configuration overlay files :file:`overlay-net-c
...
+The packets from the device are sent to the host over the net capture tunnel using IP in IP tunneling.
+The nRF70 Series device sends the IEEE 802.11 packets prepended with a custom heametadata) over the net capture tunnel to the host.
+To analyze the packets in Wireshark, the payload of the UDP packets must be dissected as IEEE 802.11 packets.
+
+This support is only available in Wireshark 4.3 (under development).
+A custom build of Wireshark from the latest sources is required.
+Once the custom build is installed, complete the following steps to dissect the payload of the UDP packets as IEEE 802.11 packets:
+
+1. Ensure Wireshark is compiled with Lua support.
+#. Open Wireshark and go to **Analyze** > **Decode As** > **+**, then select **UDP**.
```
Comment (kittydepa) [inline samples/wifi/monitor/README.rst:141]: These are buttons in Wireshark? If so, please use: 

:guilabel:`button name`

@richabp please confirm first.

File: samples/wifi/monitor/README.rst [hunk 5]
```diff
@@ -126,6 +126,61 @@ To enable this feature, use the configuration overlay files :file:`overlay-net-c
 When the offline net capture feature is enabled, incoming IEEE 802.11 packets are routed to the offline storage over the net capture tunnel.
 These packets can then be analyzed using Wireshark.
 
+Wireshark decode as IEEE 802.11
```
Comment (kittydepa) [inline samples/wifi/monitor/README.rst:129]: Is the Wireshark program something we should link to in the requirements? And/or Lua programming language?

**Review by kittydepa — CHANGES_REQUESTED** [review]

File: samples/wifi/monitor/README.rst (hunk 4 above)
Comment (richabp) [reply samples/wifi/monitor/README.rst:141]: @krish2718 to confirm as I am not familiar with Wireshark UI.

Comment (richabp) [reply samples/wifi/monitor/README.rst:141]: It seems that these are Menu and Menu items, if so, then use, :guilabel:`menu name`

Comment (krish2718) [reply samples/wifi/monitor/README.rst:141]: Yes, these are the menu navigations.

File: samples/wifi/monitor/README.rst (hunk 5 above)
Comment (krish2718) [reply samples/wifi/monitor/README.rst:129]: I can add a link https://www.wireshark.org/ and https://wiki.wireshark.org/Lua links

Comment (krish2718) [reply samples/wifi/monitor/README.rst:129]: Should I just use RST syntax or add them to `links.txt`?

Comment (krish2718) [reply samples/wifi/monitor/README.rst:131]: I went with `links.txt`

File: samples/wifi/monitor/README.rst [hunk 6]
```diff
@@ -121,11 +121,68 @@ Offline net capture
 *******************
 
 The sample supports the offline net capture feature in Zephyr, see `Zephyr net capture`_ for details.
-To enable this feature, use the configuration overlay files :file:`overlay-net-capture.conf` and :file:`overlay-netusb.conf`.
+You only need `Zephyr net capture Linux setup`_ section to set up the Linux host for offline net capture.
```
Comment (kittydepa) [inline samples/wifi/monitor/README.rst:124]:
```suggestion
See the `Zephyr net capture Linux setup`_ section for instructions on how to set up the Linux host for offline net capture.
```

File: samples/wifi/monitor/README.rst [hunk 7]
```diff
@@ -121,11 +121,68 @@ Offline net capture
...
+Please ensure the requirements from `Zephyr net capture`_ are met before proceeding.
```
Comment (kittydepa) [inline samples/wifi/monitor/README.rst:125]:
```suggestion
Ensure the requirements from `Zephyr net capture`_ are met before proceeding.
```

**Review by kittydepa — APPROVED** [review]

File: samples/wifi/monitor/README.rst [hunk 8]
```diff
@@ -121,11 +121,70 @@ Offline net capture
...
+
+This support is only available in `Wireshark`_ 4.3 (under development).
+A custom build of `Wireshark`_ from the latest sources is required.
+Once the custom build is installed, complete the following steps to dissect the payload of the UDP packets as IEEE 802.11 packets:
+
+1. Ensure Wireshark is compiled with Lua support, see `Wireshark with Lua`_ for details.
+#. Open Wireshark and go to :guilabel:`Analyze` > :guilabel:`Decode As` > :guilabel:`+`, then select :guilabel:`UDP`.
+#. In the **Current** column, ensure `IEEE 802.11` is available.
+
+     If not then `Wireshark`_ does not have the support to decode the UDP payload as IEEE 802.11 packets.
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:146]: Formatting issue:
```suggestion
#. In the **Current** column, ensure `IEEE 802.11` is available.

    If not then `Wireshark`_ does not have the support to decode the UDP payload as IEEE 802.11 packets.
```

File: samples/wifi/monitor/README.rst [hunk 9]
```diff
@@ -121,11 +121,70 @@ Offline net capture
...
+            wlan_dissector:call(payload, pinfo, tree)
+         end
+
+         -- Register the dissector
+         local netcapture_udp_port = 4242
+         local udp_port = DissectorTable.get("udp.port")
+         udp_port:add(netcapture_udp_port, nordic_raw_80211)
+
+#. Copy the Lua script to the Wireshark plugin directory.
+     The plugin directory can be found in the Wireshark preferences.
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:181]: Formatting issue:
```suggestion
#. Copy the Lua script to the Wireshark plugin directory.
    The plugin directory can be found in the Wireshark preferences.
```

**Review by richabp — CHANGES_REQUESTED** [review]
Minor formatting issues.

**Pull request merged**